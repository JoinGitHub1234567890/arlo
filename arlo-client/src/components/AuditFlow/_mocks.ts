import { IAuditBoard } from '../../types'

export const dummyBoard: IAuditBoard[] = [
  {
    id: '123',
    name: 'Audit Board #1',
    members: [],
  },
  {
    id: '123',
    name: 'Audit Board #1',
    members: [
      {
        name: 'John Doe',
        affiliation: '',
      },
      {
        name: 'Jane Doe',
        affiliation: 'LIB',
      },
    ],
  },
  {
    id: '123',
    name: 'Audit Board #1',
    members: [
      {
        name: 'John Doe',
        affiliation: '',
      },
      {
        name: 'Jane Doe',
        affiliation: 'LIB',
      },
    ],
    ballots: [
      {
        tabulator: '11',
        batch: 'Precinct 13',
        position: '313',
        status: 'NOT_AUDITED',
        vote: null,
        comment: '',
      },
      {
        tabulator: '17',
        batch: 'Precinct 19',
        position: '2112',
        status: 'NOT_AUDITED',
        vote: null,
        comment: '',
      },
      {
        tabulator: '23',
        batch: 'Precinct 29',
        position: '1789',
        status: 'NOT_AUDITED',
        vote: null,
        comment: '',
      },
      {
        tabulator: '17',
        batch: 'Precinct 19',
        position: '2112',
        status: 'NOT_AUDITED',
        vote: null,
        comment: '',
      },
      {
        tabulator: '23',
        batch: 'Precinct 29',
        position: '1789',
        status: 'AUDITED',
        vote: null,
        comment: '',
      },
    ],
  },
  {
    id: '123',
    name: 'Audit Board #1',
    members: [
      {
        name: 'John Doe',
        affiliation: '',
      },
      {
        name: 'Jane Doe',
        affiliation: 'LIB',
      },
    ],
    ballots: [
      {
        tabulator: '11',
        batch: 'Precinct 13',
        position: '313',
        status: 'AUDITED',
        vote: null,
        comment: '',
      },
      {
        tabulator: '17',
        batch: 'Precinct 19',
        position: '2112',
        status: 'AUDITED',
        vote: null,
        comment: '',
      },
      {
        tabulator: '23',
        batch: 'Precinct 29',
        position: '1789',
        status: 'AUDITED',
        vote: null,
        comment: '',
      },
      {
        tabulator: '17',
        batch: 'Precinct 19',
        position: '2112',
        status: 'AUDITED',
        vote: null,
        comment: '',
      },
      {
        tabulator: '23',
        batch: 'Precinct 29',
        position: '1789',
        status: 'AUDITED',
        vote: null,
        comment: '',
      },
    ],
  },
]

export default {
  dummyBoard,
}
