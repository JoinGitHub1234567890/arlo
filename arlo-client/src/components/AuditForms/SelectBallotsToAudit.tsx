import React, { useState } from 'react'
import { toast } from 'react-toastify'
import FormSection, { FormSectionDescription } from '../Form/FormSection'
import FormWrapper from '../Form/FormWrapper'
import FormButton from '../Form/FormButton'
import FormButtonBar from '../Form/FormButtonBar'
import { Jurisdiction, Audit } from '../../types'
import { api } from '../utilities'
import { generateOptions } from '../Form/_helpers'

interface Props {
  audit: Audit
  isLoading: boolean
  setIsLoading: (isLoading: boolean) => void
  updateAudit: () => void
  getStatus: () => Promise<Audit>
}

const SelectBallotsToAudit = (props: Props) => {
  const { audit, isLoading, setIsLoading, updateAudit, getStatus } = props
  const formOneHasData = audit && audit.contests[0]
  const formTwoHasData = audit && audit.jurisdictions && audit.jurisdictions[0]
  const manifestUploaded =
    formTwoHasData &&
    audit.jurisdictions[0].ballotManifest &&
    audit.jurisdictions[0].ballotManifest.filename &&
    audit.jurisdictions[0].ballotManifest.numBallots &&
    audit.jurisdictions[0].ballotManifest.numBatches

  const formThreeHasData =
    manifestUploaded && audit.rounds && audit.rounds.length > 0

  const [manifestCSV, setManifestCSV] = useState<File | null>()

  const [numAuditBoards, setNumAuditBoards] = useState(
    formTwoHasData && audit.jurisdictions[0].auditBoards.length
  )

  const fileInputChange = (e: any) => {
    const files: any[] = e.target.files
    if (files.length < 1) {
      return
    }
    setManifestCSV(files[0])
  }

  const onAuditBoardsChange = (e: any) => {
    setNumAuditBoards(e.target.value)
  }

  const handleSubmit = async (e: any) => {
    e.preventDefault()

    const auditBoards = Array.from(Array(numAuditBoards).keys()).map(i => {
      return {
        id: `audit-board-${i + 1}`,
        members: [],
      }
    })

    try {
      // upload jurisdictions
      const data: Jurisdiction[] = [
        {
          id: 'jurisdiction-1',
          name: 'Jurisdiction 1',
          contests: [`contest-1`],
          auditBoards: auditBoards,
        },
      ]
      setIsLoading(true)
      await api('/audit/jurisdictions', {
        method: 'POST',
        body: JSON.stringify({ jurisdictions: data }),
        headers: {
          'Content-Type': 'application/json',
        },
      }).then(
        async success => {
          const newStatus = await getStatus()
          const jurisdictionID: string = newStatus.jurisdictions[0].id

          // upload form data
          if (!manifestCSV) {
            updateAudit()
            return
          }
          const formData: FormData = new FormData()
          formData.append('manifest', manifestCSV, manifestCSV.name)
          await api(`/jurisdiction/${jurisdictionID}/manifest`, {
            method: 'POST',
            body: formData,
          })

          updateAudit()
        },
        error => {
          toast.error(error.message)
          return
        }
      )
    } catch (err) {
      toast.error(err.message)
    }
  }

  const deleteBallotManifest = async (e: any) => {
    e.preventDefault()
    try {
      const jurisdictionID: string = audit.jurisdictions[0].id
      await api(`/jurisdiction/${jurisdictionID}/manifest`, {
        method: 'DELETE',
      })
      updateAudit()
    } catch (err) {
      toast.error(err.message)
    }
  }

  return formOneHasData ? (
    <form onSubmit={handleSubmit} id="formTwo">
      <FormWrapper>
        {/* <Section>
            <SectionLabel>Estimated Sample Size</SectionLabel>
            <SectionDetail>
                Choose the initial sample size you would like to use for Round 1 of the audit from the options below.
                <div><input name="sampleSize" type="radio" value="223" onChange={e => this.inputChange(e)} /><InputLabel>223 samples (80% chance of reaching risk limit in one round)</InputLabel></div>
                <div><input name="sampleSize" type="radio" value="456" onChange={e => this.inputChange(e)} /><InputLabel>456 samples (90% chance of reaching risk limit in one round)</InputLabel></div>
            </SectionDetail>
        </Section> */}
        <FormSection
          label="Number of Audit Boards"
          description="Set the number of audit boards you with to use."
        >
          <select
            id="auditBoards"
            name="auditBoards"
            value={numAuditBoards}
            onChange={e => setNumAuditBoards(parseInt(e.target.value))}
            onBlur={onAuditBoardsChange}
          >
            {generateOptions(5)}
          </select>
        </FormSection>
        <FormSection label="Ballot Manifest">
          {manifestUploaded && audit.jurisdictions[0].ballotManifest ? ( // duplicating effect of manifestUploaded for TS
            <React.Fragment>
              <FormSectionDescription>
                <b>Filename:</b>{' '}
                {audit.jurisdictions[0].ballotManifest.filename}
              </FormSectionDescription>
              <FormSectionDescription>
                <b>Ballots:</b>{' '}
                {audit.jurisdictions[0].ballotManifest.numBallots}
              </FormSectionDescription>
              <FormSectionDescription>
                <b>Batches:</b>{' '}
                {audit.jurisdictions[0].ballotManifest.numBatches}
              </FormSectionDescription>
              {!formThreeHasData && (
                <FormButton onClick={deleteBallotManifest}>
                  Delete File
                </FormButton>
              )}
            </React.Fragment>
          ) : (
            <React.Fragment>
              <FormSectionDescription>
                Click &quot;Browse&quot; to choose the appropriate Ballot
                Manifest file from your computer
              </FormSectionDescription>
              <input
                type="file"
                accept=".csv"
                onChange={fileInputChange}
              ></input>
            </React.Fragment>
          )}
        </FormSection>
      </FormWrapper>
      {!formThreeHasData && isLoading && <p>Loading...</p>}
      {!formThreeHasData && !isLoading && (
        <FormButtonBar>
          <FormButton onClick={handleSubmit}>
            Select Ballots To Audit
          </FormButton>
        </FormButtonBar>
      )}
    </form>
  ) : (
    <div></div>
  )
}

export default React.memo(SelectBallotsToAudit)
