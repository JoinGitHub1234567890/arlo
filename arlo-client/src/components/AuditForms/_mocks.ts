import { Audit } from '../../types'

export const mockAudit: Audit = {
  name: 'Primary 2019',
  riskLimit: 10,
  randomSeed: 123123123,

  contests: [
    {
      id: 'contest-1',
      name: 'Contest 1',

      choices: [
        {
          id: 'candidate-1',
          name: 'Candidate 1',
          numVotes: 42,
        },
        {
          id: 'candidate-2',
          name: 'Candidate 2',
          numVotes: 72,
        },
      ],

      totalBallotsCast: 4200,
    },
  ],

  jurisdictions: [
    {
      id: 'adams-county',
      name: 'Adams County',
      contests: ['contest-1'],
      auditBoards: [
        {
          id: 'audit-board-1',
          members: [],
        },
        {
          id: 'audit-board-2',
          members: [],
        },
      ],
      ballotManifest: {
        filename: 'Adams_County_Manifest.csv',
        numBallots: 123456,
        numBatches: 560,
        uploadedAt: '2019-06-17 11:45:00',
      },
    },
  ],

  rounds: [
    {
      startedAt: '2019-06-17 11:45:00',
      endedAt: '2019-06-17 11:55:00',
      contests: [
        {
          id: 'contest-1',
          endMeasurements: {
            pvalue: 0.085,
            isComplete: false,
          },
          results: {
            'candidate-1': 55,
            'candidate-2': 35,
          },
          sampleSize: 25,
        },
      ],
      jurisdictions: {
        'adams-county': {
          numBallots: 15,
        },
      },
    },
  ],
}

export default mockAudit
