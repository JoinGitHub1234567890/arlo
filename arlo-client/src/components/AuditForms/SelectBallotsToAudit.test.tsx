import React from 'react'
import { render, fireEvent, waitForDomChange } from '@testing-library/react'
import toastMock from 'react-toastify'
import SelectBallotsToAudit from './SelectBallotsToAudit'
import { statusStates, ballotManifest } from './_mocks'
import apiMock from '../utilities'

jest.mock('../utilities')
jest.mock('react-toastify')

describe('SelectBallotsToAudit', () => {
  it('renders correctly', () => {
    const container = render(
      <SelectBallotsToAudit
        audit={statusStates[1]}
        isLoading={false}
        setIsLoading={jest.fn()}
        updateAudit={jest.fn()}
        getStatus={jest.fn()}
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('has radio for selecting sampleSize', () => {
    const { getByText, getByLabelText } = render(
      <SelectBallotsToAudit
        audit={statusStates[1]}
        isLoading={false}
        setIsLoading={jest.fn()}
        updateAudit={jest.fn()}
        getStatus={jest.fn()}
      />
    )

    // all options should be present
    expect(getByText('BRAVO Average Sample Number: 269 samples')).toBeTruthy()
    expect(
      getByText(
        '379 samples (80% chance of reaching risk limit and completing the audit in one round)'
      )
    ).toBeTruthy()
    expect(getByText('78 samples'))

    // correct default should be selected
    expect(
      getByLabelText('BRAVO Average Sample Number: 269 samples').hasAttribute(
        'checked'
      )
    ).toBeTruthy()
  })

  it('changes sampleSize based on audit.rounds.contests.sampleSize', () => {
    const { getByLabelText } = render(
      <SelectBallotsToAudit
        audit={statusStates[4]}
        isLoading={false}
        setIsLoading={jest.fn()}
        updateAudit={jest.fn()}
        getStatus={jest.fn()}
      />
    )

    expect(
      getByLabelText(
        '379 samples (80% chance of reaching risk limit and completing the audit in one round)'
      ).hasAttribute('checked')
    ).toBeTruthy()
  })

  it('changes number of audits', () => {
    const { getByTestId } = render(
      <SelectBallotsToAudit
        audit={statusStates[1]}
        isLoading={false}
        setIsLoading={jest.fn()}
        updateAudit={jest.fn()}
        getStatus={jest.fn()}
      />
    )

    const auditBoardInput: any = getByTestId('audit-boards')
    fireEvent.change(auditBoardInput, { target: { selected: 3 } })
    expect(auditBoardInput.selected).toBe(3)
  })

  it('submits sample size, ballot manifest, and number of audits', async () => {
    ;(apiMock as jest.Mock).mockImplementation(() => Promise.resolve())
    const getStatusMock = jest
      .fn()
      .mockImplementationOnce(() => Promise.resolve(statusStates[2])) // the POST to /audit/status after jurisdictions
    const updateAuditMock = jest
      .fn()
      .mockImplementationOnce(() => Promise.resolve(statusStates[3])) // the POST to /audit/status after manifest

    const {
      getByTestId,
      getByLabelText,
      getByText,
      queryAllByText,
      container,
    } = render(
      <SelectBallotsToAudit
        audit={statusStates[1]}
        isLoading={false}
        setIsLoading={jest.fn()}
        updateAudit={updateAuditMock}
        getStatus={getStatusMock}
      />
    )

    const manifestInput = getByTestId('ballot-manifest')
    fireEvent.change(manifestInput, { target: { files: [ballotManifest] } })

    const auditBoardInput: any = getByTestId('audit-boards')
    fireEvent.change(auditBoardInput, { target: { selected: 1 } })

    const sampleSizeInput = getByLabelText(
      '379 samples (80% chance of reaching risk limit and completing the audit in one round)'
    )
    fireEvent.click(sampleSizeInput, { bubbles: true })

    const submitButton = getByText('Select Ballots To Audit')
    fireEvent.click(submitButton, { bubbles: true })

    waitForDomChange({ container }).then(
      () => {
        expect((apiMock as jest.Mock).mock.calls.length).toBe(3)

        expect((apiMock as jest.Mock).mock.calls[0][0]).toBe(
          '/audit/sample-size'
        )
        expect((apiMock as jest.Mock).mock.calls[0][1]).toMatchObject({
          method: 'POST',
          body: {
            size: 379,
          },
          headers: {
            'Content-Type': 'application/json',
          },
        })

        expect((apiMock as jest.Mock).mock.calls[1][0]).toBe(
          '/audit/jurisdictions'
        )
        expect((apiMock as jest.Mock).mock.calls[1][1]).toMatchObject({
          method: 'POST',
          body: {
            jurisdictions: [
              {
                id: 'jurisdiction-1',
                name: 'Jurisdiction 1',
                contests: ['contest-1'],
                auditBoards: {
                  id: 'audit-board-1',
                  name: 'Audit Board #1',
                  members: [],
                },
              },
            ],
          },
          headers: {
            'Content-Type': 'application/json',
          },
        })

        expect((apiMock as jest.Mock).mock.calls[2][0]).toBe(
          '/jurisdiction/jurisdiction-1/manifest'
        )

        expect((getStatusMock as jest.Mock).mock.calls.length).toBe(1)
        expect((updateAuditMock as jest.Mock).mock.calls.length).toBe(1)

        expect(queryAllByText('Select Ballots To Audit').length).toBe(0)
      },
      error => {
        throw new Error(error)
      }
    )
  })

  it('handles api error on /audit/sample-size', async () => {
    ;(apiMock as jest.Mock)
      .mockImplementationOnce(() => Promise.reject())
      .mockImplementation(() => Promise.resolve())
    const getStatusMock = jest
      .fn()
      .mockImplementationOnce(() => Promise.resolve(statusStates[2])) // the POST to /audit/status after jurisdictions
    const updateAuditMock = jest
      .fn()
      .mockImplementationOnce(() => Promise.resolve(statusStates[3])) // the POST to /audit/status after manifest

    const {
      getByTestId,
      getByLabelText,
      getByText,
      queryAllByText,
      container,
    } = render(
      <SelectBallotsToAudit
        audit={statusStates[1]}
        isLoading={false}
        setIsLoading={jest.fn()}
        updateAudit={updateAuditMock}
        getStatus={getStatusMock}
      />
    )

    const manifestInput = getByTestId('ballot-manifest')
    fireEvent.change(manifestInput, { target: { files: [ballotManifest] } })

    const auditBoardInput: any = getByTestId('audit-boards')
    fireEvent.change(auditBoardInput, { target: { selected: 1 } })

    const sampleSizeInput = getByLabelText(
      '379 samples (80% chance of reaching risk limit and completing the audit in one round)'
    )
    fireEvent.click(sampleSizeInput, { bubbles: true })

    const submitButton = getByText('Select Ballots To Audit')
    fireEvent.click(submitButton, { bubbles: true })

    waitForDomChange({ container }).then(
      () => {
        expect(apiMock).toBeCalledTimes(3) // failure on /audit/sample-size doesn't block other calls
        expect(toastMock).toBeCalledTimes(1)

        expect(getStatusMock).toBeCalledTimes(1)
        expect(updateAuditMock).toBeCalledTimes(1)

        expect(queryAllByText('Select Ballots To Audit').length).toBe(0)
      },
      error => {
        throw new Error(error)
      }
    )
  })

  it('handles api error on /audit/jurisdictions', async () => {
    ;(apiMock as jest.Mock)
      .mockImplementationOnce(() => Promise.resolve())
      .mockImplementationOnce(() => Promise.reject())
      .mockImplementation(() => Promise.resolve())
    const getStatusMock = jest
      .fn()
      .mockImplementationOnce(() => Promise.resolve(statusStates[2])) // the POST to /audit/status after jurisdictions
    const updateAuditMock = jest
      .fn()
      .mockImplementationOnce(() => Promise.resolve(statusStates[3])) // the POST to /audit/status after manifest

    const {
      getByTestId,
      getByLabelText,
      getByText,
      queryAllByText,
      container,
    } = render(
      <SelectBallotsToAudit
        audit={statusStates[1]}
        isLoading={false}
        setIsLoading={jest.fn()}
        updateAudit={updateAuditMock}
        getStatus={getStatusMock}
      />
    )

    const manifestInput = getByTestId('ballot-manifest')
    fireEvent.change(manifestInput, { target: { files: [ballotManifest] } })

    const auditBoardInput: any = getByTestId('audit-boards')
    fireEvent.change(auditBoardInput, { target: { selected: 1 } })

    const sampleSizeInput = getByLabelText(
      '379 samples (80% chance of reaching risk limit and completing the audit in one round)'
    )
    fireEvent.click(sampleSizeInput, { bubbles: true })

    const submitButton = getByText('Select Ballots To Audit')
    fireEvent.click(submitButton, { bubbles: true })

    waitForDomChange({ container }).then(
      () => {
        expect(apiMock).toBeCalledTimes(2)
        expect(toastMock).toBeCalledTimes(1)

        expect(getStatusMock).toBeCalledTimes(0)
        expect(updateAuditMock).toBeCalledTimes(0)

        expect(queryAllByText('Select Ballots To Audit').length).toBe(1)
      },
      error => {
        throw new Error(error)
      }
    )
  })

  it('handles api error on /audit/jurisdiction/:id/manifest', async () => {
    ;(apiMock as jest.Mock)
      .mockImplementationOnce(() => Promise.resolve())
      .mockImplementationOnce(() => Promise.resolve())
      .mockImplementationOnce(() => Promise.reject())
    const getStatusMock = jest
      .fn()
      .mockImplementationOnce(() => Promise.resolve(statusStates[2])) // the POST to /audit/status after jurisdictions
    const updateAuditMock = jest
      .fn()
      .mockImplementationOnce(() => Promise.resolve(statusStates[3])) // the POST to /audit/status after manifest

    const {
      getByTestId,
      getByLabelText,
      getByText,
      queryAllByText,
      container,
    } = render(
      <SelectBallotsToAudit
        audit={statusStates[1]}
        isLoading={false}
        setIsLoading={jest.fn()}
        updateAudit={updateAuditMock}
        getStatus={getStatusMock}
      />
    )

    const manifestInput = getByTestId('ballot-manifest')
    fireEvent.change(manifestInput, { target: { files: [ballotManifest] } })

    const auditBoardInput: any = getByTestId('audit-boards')
    fireEvent.change(auditBoardInput, { target: { selected: 1 } })

    const sampleSizeInput = getByLabelText(
      '379 samples (80% chance of reaching risk limit and completing the audit in one round)'
    )
    fireEvent.click(sampleSizeInput, { bubbles: true })

    const submitButton = getByText('Select Ballots To Audit')
    fireEvent.click(submitButton, { bubbles: true })

    waitForDomChange({ container }).then(
      () => {
        expect(apiMock).toBeCalledTimes(3)
        expect(toastMock).toBeCalledTimes(1)

        expect(getStatusMock).toBeCalledTimes(1)
        expect(updateAuditMock).toBeCalledTimes(1)

        expect(queryAllByText('Select Ballots To Audit').length).toBe(1)
      },
      error => {
        throw new Error(error)
      }
    )
  })

  it('uses the highest prob value from duplicate sampleSizes', () => {
    statusStates[1].contests[0].sampleSizeOptions = [
      { size: 30, prob: 0.8 },
      { size: 30, prob: 0.9 },
    ]
    const { queryAllByText } = render(
      <SelectBallotsToAudit
        audit={statusStates[1]}
        isLoading={false}
        setIsLoading={jest.fn()}
        updateAudit={jest.fn()}
        getStatus={jest.fn()}
      />
    )

    expect(
      queryAllByText(
        '30 samples (90% chance of reaching risk limit and completing the audit in one round)'
      ).length
    ).toBe(1)
  })

  it('does not display duplicate sampleSize options', () => {
    const statusState = { ...statusStates[1] }
    statusState.contests[0].sampleSizeOptions = [{ size: 30 }, { size: 30 }]
    const { queryAllByText } = render(
      <SelectBallotsToAudit
        audit={statusState}
        isLoading={false}
        setIsLoading={jest.fn()}
        updateAudit={jest.fn()}
        getStatus={jest.fn()}
      />
    )

    expect(queryAllByText('30 samples').length).toBe(1)
  })
})
