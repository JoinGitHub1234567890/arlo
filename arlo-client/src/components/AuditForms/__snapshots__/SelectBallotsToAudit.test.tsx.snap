// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders corretly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        id="formTwo"
      >
        <div
          class="sc-ifAKCX gVocmv"
        >
          <div
            class="sc-bxivhb doxbWk"
          />
          <div
            class="sc-bdVaJa fdbLH"
          >
            <div
              class="sc-htpNat kRUApY"
            >
              Number of Audit Boards
            </div>
            <div
              class="sc-bwzfXH xctbO"
            >
              Set the number of audit boards you with to use.
            </div>
            <select
              id="auditBoards"
              name="auditBoards"
            >
              <option>
                1
              </option>
              <option>
                2
              </option>
              <option>
                3
              </option>
              <option>
                4
              </option>
              <option>
                5
              </option>
            </select>
          </div>
          <div
            class="sc-bdVaJa fdbLH"
          >
            <div
              class="sc-htpNat kRUApY"
            >
              Ballot Manifest
            </div>
            <div
              class="sc-bwzfXH xctbO"
            >
              <b>
                Filename:
              </b>
               
              Adams_County_Manifest.csv
            </div>
            <div
              class="sc-bwzfXH xctbO"
            >
              <b>
                Ballots:
              </b>
               
              123456
            </div>
            <div
              class="sc-bwzfXH xctbO"
            >
              <b>
                Batches:
              </b>
               
              560
            </div>
            <button
              class="sc-EHOje hXhzUo"
            >
              Delete File
            </button>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      id="formTwo"
    >
      <div
        class="sc-ifAKCX gVocmv"
      >
        <div
          class="sc-bxivhb doxbWk"
        />
        <div
          class="sc-bdVaJa fdbLH"
        >
          <div
            class="sc-htpNat kRUApY"
          >
            Number of Audit Boards
          </div>
          <div
            class="sc-bwzfXH xctbO"
          >
            Set the number of audit boards you with to use.
          </div>
          <select
            id="auditBoards"
            name="auditBoards"
          >
            <option>
              1
            </option>
            <option>
              2
            </option>
            <option>
              3
            </option>
            <option>
              4
            </option>
            <option>
              5
            </option>
          </select>
        </div>
        <div
          class="sc-bdVaJa fdbLH"
        >
          <div
            class="sc-htpNat kRUApY"
          >
            Ballot Manifest
          </div>
          <div
            class="sc-bwzfXH xctbO"
          >
            <b>
              Filename:
            </b>
             
            Adams_County_Manifest.csv
          </div>
          <div
            class="sc-bwzfXH xctbO"
          >
            <b>
              Ballots:
            </b>
             
            123456
          </div>
          <div
            class="sc-bwzfXH xctbO"
          >
            <b>
              Batches:
            </b>
             
            560
          </div>
          <button
            class="sc-EHOje hXhzUo"
          >
            Delete File
          </button>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
